lib/components/term.js

```diff
@@ -144,38 +146,44 @@ export default class Term extends React.PureComponent {
 
     if (!props.term) {
       this.term.attachCustomKeyEventHandler(this.keyboardHandler);
+      this.term.loadAddon(fitAddon);
+      this.term.loadAddon(searchAddon);
+      this.term.loadAddon(webLinksAddon);
       this.term.open(this.termRef);
-      this.term.webLinksInit();
-      this.term.winptyCompatInit();
+      fitAddon.fit();
+      if (this.term.useWebGL) {
+        this.term.loadAddon(webGlAddon);
+      }
+      // this.term.winptyCompatInit();
     }
 
     if (this.props.isTermActive) {
       this.term.focus();
     }
 
     if (props.onTitle) {
-      this.disposableListeners.push(this.term.addDisposableListener('title', props.onTitle));
+      this.disposableListeners.push(this.term.onTitleChange(props.onTitle));
     }
 
     if (props.onActive) {
-      this.disposableListeners.push(this.term.addDisposableListener('focus', props.onActive));
+      this.disposableListeners.push(this.term.textarea.addEventListener('focus', props.onActive));
```

Comment:
`addEventListener` doesn't returns an object with a `dispose` function, which means you shouldn't add it to `disposableListeners`.
There needs to be an additional `removeListener` statement to clear it, at around line 380.